libs/test-common/src/main/java/org/codice/ddf/test/common/mockito/PrivilegedVerificationMode.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.test.common.mockito;
+
+import static org.mockito.Mockito.times;
+
+import com.google.common.annotations.Beta;
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.internal.verification.VerificationModeFactory;
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.internal.verification.api.VerificationDataInOrder;
+import org.mockito.internal.verification.api.VerificationInOrderMode;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.Location;
+import org.mockito.invocation.MatchableInvocation;
+import org.mockito.verification.VerificationMode;
+
+/**
+ * Class that extends Mockito's {@link VerificationMode} to verify that a mocked method has been
+ * called within a {@code AccessController#doPrivileged()} block.
+ *
+ * <p><b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+@Beta
+public class PrivilegedVerificationMode implements VerificationMode, VerificationInOrderMode {
+
+  private final VerificationMode mode;
+
+  private PrivilegedVerificationMode(VerificationMode mode) {
+    this.mode = mode;
+  }
+
+  /**
+   * Wraps another {@link VerificationMode} instance and ensures all the expected calls were made
```

Comment:
✏️ Maybe something like `ensures all verified calls from the specified mode were made inside ...`. That way if one does `privileged(never())` it would succeed if nothing was invoked.